lib/publiccloud/instance.pm

```diff
@@ -919,9 +919,18 @@ sub do_systemd_analyze_time {
 
 sub upload_supportconfig_log {
     my ($self, %args) = @_;
-    $self->ssh_script_run(cmd => 'sudo supportconfig -R /var/tmp -B supportconfig -x AUDIT', timeout => 600);
-    $self->ssh_script_run(cmd => 'sudo chmod 0644 /var/tmp/scc_supportconfig.txz');
-    $self->upload_log('/var/tmp/scc_supportconfig.txz', failok => 1, timeout => 600);
+    return unless (get_var('SUPPORTCONFIG_LOG', 1));
+    # to exclude nothings, set "-"
+    my $exclude = get_var('SUPPORTCONFIG_EXCLUDE', 'AUDIT');
+    $exclude = ($exclude =~ s/[-\s]+//gr) ? '-x ' . $exclude : '';
+    my $log = "/var/tmp/scc_supportconfig";
+    my $cmd = "timeout --preserve-status 900 sudo supportconfig -R " . dirname($log) . " -B supportconfig $exclude";
+    record_info('supportconfig start', "start time: " . localtime());
+    my $err = $self->ssh_script_run(cmd => "$cmd", timeout => 960, proceed_on_failure => 1);
+    $self->ssh_script_run(cmd => "ls -l " . dirname($log) . "; [ -f $log.txz ] || sudo tar -cv -f $log.txz -J $log; sudo chmod 0644 $log.txz", proceed_on_failure => 1);
+    $cmd = ($err) ? "Stopped at " . localtime() . ", Partial Log" : "Completed at " . localtime() . ", Log";
+    $self->upload_log("$log.txz", failok => 1, timeout => 900);
+    record_info('supportconfig end', "$cmd $log.txz", result => ($err) ? 'fail' : 'ok');
```

Comment:
As per above suggestion:

```suggestion
    record_info('supportconfig end', "$msg $log.txz", result => ($err) ? 'fail' : 'ok');
```

Reply:
same reply: makes it sense to add a new variable instead of reusing existing one?

Reply:
> same reply: makes it sense to add a new variable instead of reusing existing one?

Yes.  We should use a new variable because this is a message and not a command.